spec/requests/v1/qc_reception_spec.rb

```diff
@@ -0,0 +1,266 @@
+          expect(JSON.parse(response.parsed_body)['errors'][0]['title']).to eq error_message
+        end
+      end
+
+      context 'when source is missing' do
+        let(:invalid_body) do
+          {
+            data: {
+              type: 'qc_receptions',
+              attributes: {
+                qc_results_list:
+              }
+            }
+          }.to_json
+        end
+
+        it 'does not create a new QcReception' do
+          expect do
+            post v1_qc_receptions_url, params: invalid_body, headers: json_api_headers
+          end.not_to change(QcReception, :count)
+        end
+
+        it 'renders a JSON response with errors for the new qc_receptions' do
+          post v1_qc_receptions_url, params: invalid_body, headers: json_api_headers
+          expect(response).to have_http_status(:unprocessable_entity)
+          expect(response.content_type).to match(a_string_including('application/vnd.api+json'))
+          expect(JSON.parse(response.parsed_body)['errors'][0]['detail']).to eq "source - can't be blank"
+        end
+      end
+
+      context 'when source is empty' do
+        let(:invalid_body) do
+          {
+            data: {
+              type: 'qc_receptions',
+              attributes: {
+                source: '',
+                qc_results_list:
+              }
+            }
+          }.to_json
+        end
+
+        it 'does not create a new QcReception' do
+          expect do
+            post v1_qc_receptions_url, params: invalid_body, headers: json_api_headers
+          end.not_to change(QcReception, :count)
+        end
+
+        it 'renders a JSON response with errors for the new qc_results_list' do
+          post v1_qc_receptions_url, params: invalid_body, headers: json_api_headers
+          expect(response).to have_http_status(:unprocessable_entity)
+          expect(response.content_type).to match(a_string_including('application/vnd.api+json'))
+          expect(JSON.parse(response.parsed_body)['errors'][0]['detail']).to eq "source - can't be blank"
+        end
+      end
+
+      context 'when data is missing' do
+        let(:invalid_body) do
+          {}.to_json
+        end
+
+        it 'does not create a new QcReception' do
+          expect do
+            post v1_qc_receptions_url, params: invalid_body, headers: json_api_headers
+          end.not_to change(QcReception, :count)
+        end
+
+        it 'renders a JSON response with errors for the new qc_receptions' do
+          post v1_qc_receptions_url, params: invalid_body, headers: json_api_headers
+          expect(response).to have_http_status(:bad_request)
+          expect(response.content_type).to match(a_string_including('application/vnd.api+json'))
+          expect(JSON.parse(response.parsed_body)['errors'][0]['detail']).to eq 'The required parameter, data, is missing.'
+        end
+      end
+    end
+  end
+end
```

Comment:
Thorough tests, just wondering if they can be refactored a little?

Reply:
@Sangeetha-Bheeman todo? If not, resolve the comment

Reply:
Will resolve it now, and come back to it later.